src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackTool.targets

```diff
@@ -80,14 +88,26 @@ Copyright (c) .NET Foundation. All rights reserved.
        ToolsSettingsFilePath="$(_ToolsSettingsFilePath)" />
   </Target>
 
-  <Target Name="ResolveApphostAsset" DependsOnTargets="ResolveReferences;Publish">
+  <!--
+    ============================================================
+                                        GenerateShimsAssets
+
+    Generate packaged shims for signing when PackAsToolShimRuntimeIdentifiers is set
+    ============================================================
+    -->
+
+  <Target Name="GenerateShimsAssets"
+          AfterTargets="Build"
```

Comment:
@nguerrera do you think use `AfterTargets="Build"` should be the trigger for generating shim during build? I don't think there is a way to make build depends on this task